crates/edr_napi/src/solidity_tests/config.rs

```diff
@@ -234,6 +242,24 @@ impl SolidityTestRunnerConfigArgs {
 
         let fuzz: FuzzConfig = fuzz.map(TryFrom::try_from).transpose()?.unwrap_or_default();
 
+        // Collect the functions that allow internal expect revert
+        let functions_internal_expect_revert: HashSet<TestFunctionIdentifier> =
+            test_function_overrides
+                .as_ref()
+                .map(|overrides| {
+                    overrides
+                        .iter()
+                        .filter(|override_item| {
+                            override_item
+                                .config
+                                .allow_internal_expect_revert
+                                .unwrap_or_default()
+                        })
+                        .map(|override_item| override_item.identifier.clone())
+                        .collect()
+                })
+                .unwrap_or_default();
```

Comment:
Question: Do you need the N-API object representation of `TestFunctionIdentifier`? Alternatively, could you combine this with:

```rs
            functions_internal_expect_revert: functions_internal_expect_revert
                .into_iter()
                .map(TryInto::try_into)
                .collect::<Result<_, _>>()?,
```

to avoid allocating the `HashSet` twice?

If so, that would also void the need for (most of) these [derivations](https://github.com/NomicFoundation/edr/pull/1239/files#diff-70a22b4cfb5cd42aca82d1fd1e8c8c3bb8a0f0a016da7a18bd8c16b7dcbe0bdeR19) on `ArtifactId` and `TestFunctionIdentifier`